core/src/main/java/io/grpc/internal/ClientStream.java

```diff
@@ -72,6 +73,7 @@
    * <p>This method should not throw any exceptions.
    *
    * @param listener non-{@code null} listener of stream events
+   * @param headers to send at the beginning of the call
    */
-  void start(ClientStreamListener listener);
+  void start(ClientStreamListener listener, Metadata headers);
```

Comment:
This PR can't be accepted as-is, because it has broken `Metadata` usage. After `Metadata` is passed to the transport it can't be read/modified any further by the Call, because the transport will be read/modifying it. That's why I wanted it to be `null`ed out before, because after being passed to the transport any use of the `Metadata` by Call is a bug.